docs/content/practices/network.md

```diff
@@ -1,5 +1,148 @@
+* **Production server**
+  <br>_You will get 100% `E2E`. On the other hand, not all applications will be able to do this (for example, in the banking community)_
+
+* **Development server**
+  <br>_We still have `E2E`, but the environment can be significantly different from production. Also, regular redeployment
+  can be a problem_
+* **Special stage server**
+  <br>_Can be different with your production/development backend. Also pay your attention, that it requires an
+  additional effort from engineers to support it. It also is not 100% end to end, as it tests not real backend side._
+  
+???+ warning
+    It doesn't matter which of the options above you choose: Internet connection always can be an issue. 
+    <br>If you use UI testing as a part of pull request or release pipeline, it's better to avoid real network usage there, but just run them regular several times a day 
+
+### Mock network (manually)
+
+* **[MockWebServer](https://github.com/square/okhttp/tree/master/mockwebserver)**
+  <br>_Part of `OkHttp` library. Allows intercepting and responding of each http request._
+  <br>You can find more details about it [here](https://adambennett.dev/2021/09/mockwebserver-https/)
+  
+* **Custom Interceptor**
+  <br>_In the simplest implementation, this is a common interceptor which is intercepts each http call, which is added only in tests and actually replaces
+  triggered requests for predefined responses. 
+  <br> However, unlike `OkHttpWebServer`, it works for more
+  high level, therefore, may miss some bugs. 
+  <br>Also, in the most of the cases, we don't need to test third party libraries code_
+* **DI**
+  <br> _Each DI framework ([Dagger2](https://github.com/google/dagger) / [Toothpick](https://github.com/stephanenicolas/toothpick) / [Koin](https://github.com/InsertKoinIO/koin) / etc) allows you to replace dependencies in tests.
+  <br> We can replace the `Retroift Service` or the whole` Repository` entirely. However, we actually crop
+  testing the entire network layer and offline mod if it exists_
+
+**Problems**
+
+You need to mock manually. Imagine, that in your flow has hundreds of http requests.
+<br> You may spend a lot of time to have it done, and it's may be really challenging to keep these tests up-to-dated.
+
+### Mock network (automatically)
+
+The key idea is that two modes of network operation appear in the tests:`Record & Replay`.
+
+* **Record**
+  <br>_In this mode, the test uses a real backend. However, it also records all requests and exports them to set or to one file that can be associated with a specific test. 
+  <br>This mode is used locally only when writing a test_
+
+* **Replay**
+  <br>_Instead of a real network, an already recorded file is used. This mode is used both locally and on CI. It allows to fully exclude all the possible problems with a network.
+  Also, we've got an opportunity to easily mock authorisation in this mode, as we've already had all the data we need to put into preferences/stores_
```

Comment:
typo: as we already have

Reply:
Also not very obvious what this "all the data we need to put into preferences/stores" means.

I suggest something like:
We already have retrieved user's profile data. This data is likely stored locally in disk. In doing so, we can save in http requests that require such infos by fetching user's profile data swiftly from the disk instead

Reply:
fixed